sr_utilities_common/scripts/sr_utilities_common/performance_test.py

```diff
@@ -22,13 +24,42 @@ def __init__(self):
 
         rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self._diagnostics_agg_callback)
         if self._hand_e:
-            rospy.Subscriber("/rh/debug_etherCAT_data", EthercatDebug, self._ethercat_data_callback)
-
-        self.total_control_loop_overruns_count = 0
-        self.total_recent_control_loop_overruns_count = 0
-        self.total_invalid_packets_total_count = list()
-        self.total_invalid_packets_recent = 0
-        self._invalid_hand_e_packets_total = 0
+            rospy.Subscriber("/rh/debug_etherCAT_data", EthercatDebug, self._ethercat_data_hand_e_callback)
+
+        self.total_control_loop_overruns = 0
+        self.max_recent_control_loop_overruns = 0
+        self.total_invalid_packets = list()
+        self.max_invalid_packets_recent = list()
+        self.invalid_hand_e_packets_total = 0
+        self._start_hand_e_ethercat_time = 0.0
+        self._old_invalid_hand_e_packets = 0
+        self.max_invalid_hand_e_packets_recent = 0
+        self._total_control_loop_overruns_threshold = 0
+        self._max_recent_control_loop_overruns_threshold = 0
+        self._total_invalid_packets_threshold = 0
+        self._max_invalid_packets_recent_threshold = 0
+
+        self._parse_thresholds_from_yaml()
+
+    def _parse_thresholds_from_yaml(self):
+        rospack = rospkg.RosPack()
+        yaml_path = rospack.get_path('sr_utilities_common') + '/config/yaml/performance_script_thresholds.yaml'
```

Comment:
Please load the thresholds to the parameter server as local parameters in the launchfile.
Then read the parameters from the parameter server.
Instead of reading the yaml directly.